examples/hello/client.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+#include "hello.h"
+
+/*
+ * this file represent a client sendind a message to a specified server
+ */
+
+int main(int argc, char **argv)
+{
+	int ret;
+	ptl_handle_ni_t retInit;
+	ptl_handle_eq_t retEQAlloc;
+	ptl_handle_md_t resBind;
+	ptl_event_t retWait;
+	ptl_hdr_data_t header_data = 1234;
+	ptl_event_t reseqpoll;
+	char msg[PTL_EV_STR_SIZE];
+	char *p;
+
+	if (argc < 3) {
+		fprintf(stderr, "Usage: client <nid> <message>\n not enough arguments");
+		return 1;
+	}
+
+	int nid = strtol(argv[1], &p, 10);
+	if (*p != '\0') {
+		fprintf(stderr, "An invalid character was found before the end of the string");

```

Comment:
Missing an '\n'
Please declare the variable before, at the start of the function. It should also be a `long`, and we should check that it is not larger than PTL_NID_MAX